astroquery/ipac/irsa/most/core.py

```diff
@@ -0,0 +1,1017 @@
+    'https://irsa.ipac.caltech.edu/workspace/TMP_X69utS_13312/MOST/pid15792/fitsimage_A850RA.tar.gz'
+    >>> matched["region_tarball"]  # doctest: +IGNORE_OUTPUT
+    'https://irsa.ipac.caltech.edu/workspace/TMP_X69utS_13312/MOST/pid15792/ds9region_A850RA.tar'
+    >>> matched["results"].columns  # doctest: +IGNORE_OUTPUT
+    <TableColumns names=('Image_ID','date_obs','time_obs','mjd_obs','ra_obj','dec_obj','sun_dist','geo_dist','dist_ctr','phase','vmag','image_url','postcard_url','region_file')>  # noqa: E501
+    >>> matched["results"]["time_obs", "image_url"]  # doctest: +IGNORE_OUTPUT
+    <Table length=10>
+      time_obs                                                  image_url
+       str12                                                      str103
+    ------------ -------------------------------------------------------------------------------------------------------
+    11:00:08.319 https://irsa.ipac.caltech.edu/ibe/data/wise/merge/merge_p1bm_frm/3b/49273b/134/49273b134-w2-int-1b.fits
+    11:00:08.319 https://irsa.ipac.caltech.edu/ibe/data/wise/merge/merge_p1bm_frm/3b/49273b/134/49273b134-w1-int-1b.fits
+    14:09:44.351 https://irsa.ipac.caltech.edu/ibe/data/wise/merge/merge_p1bm_frm/7b/49277b/135/49277b135-w1-int-1b.fits
+    14:09:44.351 https://irsa.ipac.caltech.edu/ibe/data/wise/merge/merge_p1bm_frm/7b/49277b/135/49277b135-w2-int-1b.fits
+    17:19:09.391 https://irsa.ipac.caltech.edu/ibe/data/wise/merge/merge_p1bm_frm/1b/49281b/134/49281b134-w2-int-1b.fits
+    17:19:09.391 https://irsa.ipac.caltech.edu/ibe/data/wise/merge/merge_p1bm_frm/1b/49281b/134/49281b134-w1-int-1b.fits
+    20:28:45.431 https://irsa.ipac.caltech.edu/ibe/data/wise/merge/merge_p1bm_frm/5b/49285b/135/49285b135-w2-int-1b.fits
+    20:28:45.431 https://irsa.ipac.caltech.edu/ibe/data/wise/merge/merge_p1bm_frm/5b/49285b/135/49285b135-w1-int-1b.fits
+    23:38:10.476 https://irsa.ipac.caltech.edu/ibe/data/wise/merge/merge_p1bm_frm/9b/49289b/134/49289b134-w1-int-1b.fits
+    23:38:10.476 https://irsa.ipac.caltech.edu/ibe/data/wise/merge/merge_p1bm_frm/9b/49289b/134/49289b134-w2-int-1b.fits
+
+.. note::
+    The returned TAR Archives point to the MOST service hosted directory which
+    will dissapear after a while, making the URLs return a 404 Not Found Error.
+    The URLs returned by the results table, however, point to the NASA/IPAC
+    Infrared Science Archive, which means that the URLs to the images themselves
+    will remain valid even after the MOST URLs expire.
+"""
+import io
+import re
+import tarfile
+
+from bs4 import BeautifulSoup
+
+from astropy.io import votable, fits
+
+from astropy.table import Table
+
+from astroquery.query import BaseQuery
+from astroquery.utils import class_or_instance
+
+from . import conf
+
+
+__all__ = ["Most", "MOSTClass"]
+
+
+class MOSTClass(BaseQuery):
+    URL = conf.server
+    TIMEOUT = conf.timeout
+
+    def __init__(self):
+        super().__init__()
+
+    def _validate_name_input_type(self, params):
```

Comment:
~~these methods a little bit overkill as they are being used only once and they do a one liner, but now that we have it, they could stay.~~

OK, I take it back, I see that the mpc or manual inputs are a bit more complicated. I leave the previous comment here though in case someone else falls into the same review train of thought :)

Reply:
Yes, the methods are broken out for consistency and clarity more than for necessity. Too many deeply nested `if`s.